packages/apollo-gateway/src/loadServicesFromRemoteEndpoint.ts

```diff
@@ -22,31 +26,34 @@ export async function getServiceDefinitionsFromRemoteEndpoint({
   let isNew = false;
   // for each service, fetch its introspection schema
   const services: ServiceDefinition[] = (await Promise.all(
-    serviceList.map(service => {
-      if (!service.url) {
-        throw new Error(
-          `Tried to load schema from ${service.name} but no url found`,
-        );
-      }
-      return fetch(service.url, {
-        method: 'POST',
-        body: JSON.stringify({
-          query: 'query GetServiceDefinition { _service { sdl } }',
-        }),
-        headers: { 'Content-Type': 'application/json', ...headers },
-      })
-        .then(res => res.json())
-        .then(({ data, errors }: GraphQLExecutionResult) => {
+    serviceList.map(({ serviceDefinition, dataSource }) => {
+      const request: GraphQLRequest = {
+        query: 'query GetServiceDefinition { _service { sdl } }',
+        http: {
+          url: <string>serviceDefinition.url,
+          method: 'POST',
+          headers: new Headers(headers),
+        },
+      };
+
+      return dataSource
+        .process({ request, context: {} })
+        .then(({ data, errors }: GraphQLResponse) => {
```

Comment:
Nit: TS should be able to infer the `GraphQLResponse` type based on the return type of `RemoteGraphQLDataSource.process()`.